--- /usr/share/livecd-rootfs/live-build/auto/config.orig
+++ /usr/share/livecd-rootfs/live-build/auto/config
@@ -832,7 +832,7 @@
 						OPTS="${OPTS:+$OPTS }--initramfs=none"
 						OPTS="${OPTS:+$OPTS }--system=normal"
 						OPTS="${OPTS:+$OPTS }--hdd-label=cloudimg-rootfs"
-						OPTS="${OPTS:+$OPTS }--ext-resize-blocks=536870912 --ext-block-size=4096"
+						OPTS="${OPTS:+$OPTS }--ext-resize-blocks=$((536870912 * 2)) --ext-block-size=4096"
 						OPTS="${OPTS:+$OPTS }--ext-fudge-factor=15"
 						;;
 					intel-iot)
@@ -840,7 +840,7 @@
 						OPTS="${OPTS:+$OPTS }--initramfs=none"
 						OPTS="${OPTS:+$OPTS }--system=normal"
 						OPTS="${OPTS:+$OPTS }--hdd-label=cloudimg-rootfs"
-						OPTS="${OPTS:+$OPTS }--ext-resize-blocks=536870912 --ext-block-size=4096"
+						OPTS="${OPTS:+$OPTS }--ext-resize-blocks=$((536870912 * 2)) --ext-block-size=4096"
 						OPTS="${OPTS:+$OPTS }--ext-fudge-factor=15"
 						;;
 					*)
@@ -1272,7 +1272,7 @@
 		esac
 		OPTS="${OPTS:+$OPTS }--system=normal"
 		OPTS="${OPTS:+$OPTS }--hdd-label=cloudimg-rootfs"
-		OPTS="${OPTS:+$OPTS }--ext-resize-blocks=536870912 --ext-block-size=4096"
+		OPTS="${OPTS:+$OPTS }--ext-resize-blocks=$((536870912 * 2)) --ext-block-size=4096"
 		OPTS="${OPTS:+$OPTS }--ext-fudge-factor=15"
 		;;
 
@@ -1598,9 +1598,11 @@
 			esac
 		fi
 
-		if [ "$IMAGEFORMAT" = none ]; then
-			rm -f config/hooks/*.binary*
-		fi
+                if [ "$IMAGEFORMAT" = none ]; then
+                        if [ "$IMAGE_FORCE_HOOKS" != "true" ]; then
+                                rm -f config/hooks/*.binary*
+                        fi
+                fi
 		;;
 esac
 
