--- /usr/lib/live/build/lb_chroot_apt.orig
+++ /usr/lib/live/build/lb_chroot_apt
@@ -47,6 +47,8 @@
 		then
 			echo "Acquire::ftp::Proxy \"${LB_APT_FTP_PROXY}\";" > chroot/etc/apt/apt.conf.d/00ftp-proxy
 		fi
+
+		echo 'Acquire::Retries "20";' > chroot/etc/apt/apt.conf.d/99retries
 
 		# Configuring apt http proxy
 		if [ -n "${LB_APT_HTTP_PROXY}" ]
@@ -237,6 +239,8 @@
 		# Deconfiguring aptitude ftp proxy
 		rm -f chroot/etc/apt/apt.conf.d/00ftp-proxy
 
+		rm -f chroot/etc/apt/apt.conf.d/99retries
+
 		# Deconfiguring aptitude http proxy
 		rm -f chroot/etc/apt/apt.conf.d/00http-proxy
 
